### PR TITLE
We shall not search in case we only want to get all messages

### PR DESCRIPTION
**Pull request by DeepDiver1975**
refs #664 
refs #533 

Please test @Azelphur @clementhk @wurstchristoph @irgendwie @Xenopathic 

Please note searching itself will remain unchanged and bad performance is still expected.

### CODE REVIEW

**Comment by irgendwie** [comment]
> C: 6 FETCH 4094:4114 (ENVELOPE FLAGS RFC822.SIZE UID INTERNALDATE BODYSTRUCTURE BODY.PEEK[HEADER.FIELDS (IMPORTANCE LIST-POST X-PRIORITY CONTENT-TYPE)])
S: 6 OK Success
-- Command 6 took 0.1766 seconds.
C: 7 LOGOUT
S: * BYE LOGOUT Requested
S: 7 OK 73 good day (Success)
-- Command 7 took 0.1807 seconds.

Seems like the fetch doesn't get any messages for my big folder (~4000 mails in it).
Still works for folders with less mails in it (tested with folders containing up to 90 mails).

Any idea @DeepDiver1975?

**Comment by DeepDiver1975** [comment]
Grrrr .... let me have a look.

**Comment by DeepDiver1975** [comment]
@irgendwie can I ask you to send a status call to the server for that big folder? Would be interesting to see the response - THX

**Comment by jancborchardt** [comment]
@Xenopathic please test this as it tackles an issue you reported.

@irgendwie more details please. :)

**Comment by jancborchardt** [comment]
@Azelphur please also test this as it tackles your issue https://github.com/owncloud/mail/issues/664

**Comment by Gomez** [comment]
@DeepDiver1975 Whats the plan here?

**Comment by ChristophWurst** [comment]
> searching itself will remain unchanged and bad performance is still expected

Not necessarily, it's actually about twice as fast with my email provider.

File: lib/mailbox.php [hunk 1]
```diff
@@ -89,23 +89,31 @@ public function __construct($conn, $mailBox, $attributes, $delimiter='/') {
...
 
 	public function getMessages($from = 0, $count = 2, $filter = '') {
-		if ($filter instanceof Horde_Imap_Client_Search_Query) {
-			$query = $filter;
+
+		if (empty($filter)) {
+			$total = $this->getTotalMessages();
+			$from = $total - $from;
+			$to = max($from - $count + 1, 1);
+			$ids = new \Horde_Imap_Client_Ids("$from:$to", true);
```
Comment (ChristophWurst) [inline lib/mailbox.php]: This won't work in all cases as sorting is not guaranteed AFAIK

**Comment by ChristophWurst** [comment]
This sounds like a reasonable solution:
http://stackoverflow.com/questions/1387999/is-it-possible-to-use-imap-paging
http://stackoverflow.com/questions/1387999/is-it-possible-to-use-imap-paging#comment1228442_1388012

**Comment by Gomez** [comment]
Is this fixed with #1486

**Comment by ChristophWurst** [comment]
yes, #1486 replaced this PR more or less

**Pull request closed without merging**